src/mpi/mpi_transceiver.cpp

```diff
@@ -20,13 +20,14 @@
 
 void mpi_transceiver::init()
 {
-    int is_initialized;
-    MPI_Initialized(&is_initialized);
+    int is_mpi_initialized = 0;
+    MPI_Initialized(&is_mpi_initialized);
     // protect against double-init
-    if(!is_initialized) {
+    if(!is_mpi_initialized) {
         MPI_Init(NULL, NULL);
-        transceiver_impl::init();
     }
+    if (!transceiver_impl::is_initialized())
```

Comment:
sorry, I was unclear, I would do the check in transceiver_impl and always do the call here. It's a bit cleaner and less error prone for other concrete implementations (to come).

Reply:
Ok, I will make the change.